code/modules/hydroponics/fermenting_barrel.dm

```diff
@@ -9,86 +9,147 @@
-
-/obj/structure/fermenting_barrel/proc/makeWine(obj/item/food/grown/fruit)
-	if(fruit.reagents)
-		fruit.reagents.trans_to(src, fruit.reagents.total_volume)
-	var/amount = fruit.seed.potency / 4
-	if(fruit.distill_reagent)
-		reagents.add_reagent(fruit.distill_reagent, amount)
+	if(open)
+		var/fruit_count = contents.len
+		if(fruit_count)
+			. += span_notice("It contains [fruit_count] fruit\s ready to be fermented.")
+		. += span_notice("It is currently open, letting you fill it with fruits or reagents.")
 	else
-		var/data = list()
-		data["names"] = list("[initial(fruit.name)]" = 1)
-		data["color"] = fruit.filling_color
-		data["boozepwr"] = fruit.wine_power
-		if(fruit.wine_flavor)
-			data["tastes"] = list(fruit.wine_flavor = 1)
-		else
-			data["tastes"] = list(fruit.tastes[1] = 1)
-		reagents.add_reagent(/datum/reagent/consumable/ethanol/fruit_wine, amount, data)
-	qdel(fruit)
-	playsound(src, 'sound/effects/bubbles.ogg', 50, TRUE)
+		. += span_notice("It is currently closed, letting it ferment fruits or draw reagents from its tap.")
 
-/obj/structure/fermenting_barrel/attackby(obj/item/I, mob/user, params)
-	var/obj/item/food/grown/fruit = I
-	if(istype(fruit))
-		if(!fruit.can_distill)
-			to_chat(user, span_warning("You can't distill this into anything..."))
-			return TRUE
-		else if(!user.transferItemToLoc(I,src))
-			to_chat(user, span_warning("[I] is stuck to your hand!"))
-			return TRUE
-		to_chat(user, span_notice("You place [I] into [src] to start the fermentation process."))
-		addtimer(CALLBACK(src, PROC_REF(makeWine), fruit), rand(80, 120) * speed_multiplier)
-		return TRUE
-	if(I)
-		if(I.is_refillable())
-			return FALSE //so we can refill them via their afterattack.
-	else
-		return ..()
+/obj/structure/fermenting_barrel/attackby(obj/item/object, mob/user, params)
+	if(open)
+		if(istype(object, /obj/item/food/grown) && insert_fruit(user, object))
+			balloon_alert(user, "Added fruit")
+			return
+		if(istype(object, /obj/item/storage/bag/plants))
+			var/obj/item/storage/bag/plants/bag = object
+			var/inserted_fruits = 0
+			for(var/obj/item/food/grown/fruit in bag.contents)
+				if(!insert_fruit(user, fruit, bag))
+					break
+				inserted_fruits++
+			if(inserted_fruits)
+				balloon_alert(user, "Added [inserted_fruits] fruit\s")
+	else if(object.is_refillable())
+		return //so we can refill them via their afterattack.
+	return ..()
 
 /obj/structure/fermenting_barrel/attack_hand(mob/user, list/modifiers)
 	if(!can_open)
 		return
-	open = !open
 	if(open)
-		reagents.flags &= ~(DRAINABLE)
-		reagents.flags |= REFILLABLE | TRANSPARENT
-		to_chat(user, span_notice("You open [src], letting you fill it."))
-	else
+		open = FALSE
 		reagents.flags |= DRAINABLE
 		reagents.flags &= ~(REFILLABLE | TRANSPARENT)
-		to_chat(user, span_notice("You close [src], letting you draw from its tap."))
-	update_appearance()
+		balloon_alert(user, "Closed the lid")
```

Comment:
I actually don't think you need either of these balloon alerts (or to_chats) since the icon change is obvious